client/src/searchPage/searchPageController.js

```diff
@@ -28,22 +30,50 @@ export default angular.module('searchPage')
                         acc[currentChunk].push(item);
                         return acc;
                     }, []);
+                    const availableFilters = [];
+                    this.filtersArr.forEach((filter) => {
+                        if (!this.filtersCount[filter]) {
+                            return;
+                        }
+                        availableFilters.push(filter);
+                    });
+                    const availableFiltersStr = availableFilters.join(',');
+                    if (availableFiltersStr && availableFiltersStr !== this.filters) {
+                        $state.go('search', {
+                            filters: availableFiltersStr,
+                        });
+                    }
+                }).catch((err) => {
+                    this.results = [];
+                    this.paginatedDocuments = [];
+                    if (err.data.filtersCount === null) {
+                        this.filtersCount = {};
+                        return;
+                    }
+                    this.filtersCount = err.data.filtersCount || this.filtersCount;
+                    if (this.filters === '') {
+                        return;
+                    }
+                    $state.go('search', {
+                        filters: '',
+                    });
                 });
             };
             filterPanelService.getFilters().then((response) => {
                 const filterGroups = response.data;
                 this.filterGroups = filterGroups;
-                searchDocuments(query, filtersArr);
+                searchDocuments(this.query, this.filtersArr, true);
                 $transitions.onRetain({
                     retained: 'search',
                 }, (transition) => {
                     const params = transition.params();
-                    const { query, filters } = params;
-                    if (this.query !== query || this.filters !== filters) {
-                        this.query = query;
-                        this.filters = filters;
-                        const filtersArr = (filters && filters.split(',').map(Number)) || [];
-                        searchDocuments(query, filtersArr);
+                    const { query: newQuery, filters: newFilters } = params;
+                    if (this.query !== newQuery || this.filters !== newFilters) {
+                        this.filters = newFilters;
+                        this.filtersArr = parseFilters(this.filters);
+                        const requestFiltersCount = this.query !== newQuery;
```

Comment:
the same for requestFiltersCount

Reply:
something like is or shouldRequestFiltersCount or shouldGetFiltersCount, something more talking about the goal and showing that it is a boolean

Reply:
fixed